### PR TITLE
Migrate base image to BCI image

### PR DESCRIPTION
**Pull request by bk201**
Migrate base image to BCI image

**Related Issue:**
https://github.com/harvester/harvester/issues/1868

**Test plan:**

### CODE REVIEW

**Review by guangbochen — APPROVED** [review]
LGTM

File: package/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,9 @@
-FROM alpine
-RUN apk update && apk add -u --no-cache git curl unzip tar tini bash nfs-utils && \
```
Comment (weihanglo) [inline package/Dockerfile:2]: Why remove nfs-utils? Is nfs utility built-in for bci-base image?

Comment (bk201) [reply package/Dockerfile:2]: https://github.com/harvester/harvester/pull/1909
It should be safe to remove nfs-utils since Harvester doesn't use it.

Comment (guangbochen) [reply package/Dockerfile:2]: Actually, the `nfs-utils` is still needed as we just move the dependency library to https://github.com/longhorn/backupstore, and the nfs mount within the container will still require to use this library.

Comment (bk201) [reply package/Dockerfile:2]: Thanks, I add the corresponding package `nfs-client`: https://github.com/harvester/harvester/compare/75cc5c6fd7b1b6ac25b0256106a24532d7bec41a..47630964c440f93cb9148a6044246bb7eb129b64

**Review by weihanglo — APPROVED** [review]
LGTM

It will be better if we can document why the package is needed. Anyhow, this could be done in the follow-up PR if the timeframe is not that tight.

**Comment by bk201** [comment]
> LGTM
> 
> It will be better if we can document why the package is needed. Anyhow, this could be done in the follow-up PR if the timeframe is not that tight.

Comments added: https://github.com/harvester/harvester/pull/1910/files#diff-c55d1ee74f32cd4b9eeeceaec981a180f52e761acb158a5b0b0e088c7725c9caR3 Thanks.

**Review by weihanglo — APPROVED** [review]
LGTM

**Review by guangbochen — APPROVED** [review]

**Pull request merged**